From 0b9a8134c361c841d3a1825a7884eaa1861444f8 Mon Sep 17 00:00:00 2001
From: David Gibson <dgibson@redhat.com>
Date: Mon, 15 Sep 2014 07:13:34 -0400
Subject: [powerpc] Revert: Add iommu domain pointer to device archdata

Message-id: <1410765214-16377-79-git-send-email-dgibson@redhat.com>
Patchwork-id: 94777
O-Subject: [PATCH 78/78] RHEL only: Revert "[powerpc] Add iommu domain pointer to device archdata"
Bugzilla: 1123145 1123133 1123367
RH-Acked-by: Radim Krcmar <rkrcmar@redhat.com>
RH-Acked-by: Don Zickus <dzickus@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>

This reverts commit 80b7421fa6211d748d87c67c7b8a406e5cd9526e.

This commit came from upstream and introducesa new field to the arch
specific portion of struct device for ppc64.  Although the name suggests
it's needed on anything with an iommu, or at least anything with VFIO, in
fact upstream it's only used for the Freescale PAMU, which exists only on
embedded platforms we don't support.

Furthermore, the extra field horribly breaks kABI when we enable IOMMU_API
to allow VFIO passthrough for upcoming KVM on Power.

Therefore, revert this patch in RHEL7.

Signed-off-by: David Gibson <dgibson@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/include/asm/device.h b/arch/powerpc/include/asm/device.h
index 38faede..77e97dd 100644
--- a/arch/powerpc/include/asm/device.h
+++ b/arch/powerpc/include/asm/device.h
@@ -28,9 +28,6 @@ struct dev_archdata {
   void  *iommu_table_base;
  } dma_data;
 
-#ifdef CONFIG_IOMMU_API
- void   *iommu_domain;
-#endif
 #ifdef CONFIG_SWIOTLB
  dma_addr_t  max_direct_dma_addr;
 #endif
-- 
1.7.1